spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/scoped/ScopedLoadBalancer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.loadbalancer.impl.scoped;
+
+import org.springframework.cloud.client.ServiceInstance;
+
+/**
+ * Implemented by classes which use a load balancer to choose a server to
+ * send a request to.
+ *
+ * @author Ryan Baxter
+ */
+public interface ScopedLoadBalancer {
```

Comment:
Have you considered doing this as a generic API that also includes quality of service?  Incorporating request lifecycle into the load balancer API would allow for more advanced load balancing strategies that take quality of service into account.  Many libraries fall short on the load balancing API by only focusing on the server selection but don't provide a feedback mechanism for more advanced load balancing algorithm.  For example, a least loaded load balancing algorithm would need to know when an operation completed so it can decrement the pending request count.  A latency aware algorithm would need to collect latency samples to update internal statistics. 

For example,

```java
public interface LoadBalancer<T> {
    enum Status {
        SUCCESSS,  // Request was handled successfully
        FAILED, // Request reached the server but failed due to timeout or internal error
        DISCARD, // Request did not go off box and should not be counted for statistics
    }

    // Context created for each request.
    interface Context {
        boolean hasServer();

        T getServer();

        // Notification that the request completed
        void complete(Status status);
    }

     // Choose the next server based on the load balancing algorithm
     Context<T> choose();
}
```

Regarding using generics, it would be much easier to test a new algorithm without having to tie it to a specific service instance API.